docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+  class MixedController extends superClass implements FindByTitleInterface<M> {
+    constructor(...args: any[]) {
+      super(...args);
+    }
+
+    @get(parms.basePath + '/findByTitle/{title}', {
+      responses: {
+        '200': {
+          description: `Array of ${parms.modelClassName} model instances`,
+          content: {
+            'application/json': {
+              schema: {
+                type: 'array',
+                items: getModelSchemaRef(parms.modelClass, {
+                  includeRelations: true,
+                }),
+              },
+            },
+          },
+        },
+      },
+    })
+    async findByTitle(@param.path.string('title') title: string): Promise<[M]> {
+      let foundItems = await this.repository.findByTitle(title);
+
+      return foundItems;
+    }
+  }
+
+  return MixedController;
+}
+```
+
+To customize certain portions of the OpenAPI description of the endpoint, the
+mixin class factory function needs to accept some input parameters. We defined
+an interface `FindByTitleControllerMixinParms` to allow for this.
+
+It is also a good idea to give the injected repository (in the controller super
+class) a generic name like `this.respository` to keep things simple in the mixin
+class factory function.
+
+### Generating A Controller Via The CLI
+
+A CLI-generated **controller** for the model `Note` would look like this: (To
+save space, only a **partial** implementation is shown)
+
+{% include code-caption.html content="src/controllers/note.controller.ts" %}
+
+```ts
+export class NoteController {
+  constructor(
+    @repository(NoteRepository)
+    public noteRepository: NoteRepository,
+  ) {}
+
+  @post('/notes', {
+    responses: {
+      '200': {
+        description: 'Note model instance',
+        content: {'application/json': {schema: getModelSchemaRef(Note)}},
+      },
+    },
+  })
+  async create(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {
+            title: 'NewNote',
+            exclude: ['id'],
+          }),
+        },
+      },
+    })
+    note: Omit<Note, 'id'>,
+  ): Promise<Note> {
+    return this.noteRepository.create(note);
+  }
+
+  // ...
+  // remaining CRUD endpoints
+  // ...
+}
+```
+
+For a full example of a CLI-generated controller for a model `Todo`, see
+[TodoController ](https://github.com/strongloop/loopback-next/blob/master/examples/todo/src/controllers/todo.controller.ts).
+
+### Adjust NoteController To Use FindByTitleControllerMixin
+
+Before we use the mixin class factory function
+
+- remove the class' `export` directive
+- rename this class to `TempNoteController`
+- change the name of the injected repository from `noteRepository` to
+  `repository` to keep things simple for the mixin class factory function
+
+{% include code-caption.html content="src/controllers/note.controller.ts" %}
+
+```ts
+class TempNoteController {
```

Comment:
Same comments as above:

1. Don't use `Temp` in the name, find a better prefix/postfix please. E.g. `NoteControllerBase` or `BaseNoteController`.
2. Even better, don't create any temporary classes, apply the mixin on the base class of the controller. In this particular case, I think you should use `Object` as the base.

```ts
class NoteController extends extends FindByTitleControllerMixin<
   Note,
   Constructor<TempNoteController>
 >(Object, mixinParms) {}
```